src/py/aspen/app/views/sample.py

```diff
@@ -29,15 +24,21 @@ def _format_created_date(sample: Sample) -> str:
         return "not yet uploaded"
 
 
-def _format_gisaid_accession(sample: Sample) -> str:
+def _format_gisaid_accession(
+    sample: Sample, entity_id_to_accession_map: Mapping[int, Accession]
+) -> str:
     if sample.uploaded_pathogen_genome is not None:
-        for accession in sample.uploaded_pathogen_genome.accessions:
-            if accession.repository_type == PublicRepositoryType.GISAID:
-                return accession.public_identifier
+        accession = entity_id_to_accession_map.get(
+            sample.uploaded_pathogen_genome.entity_id, None
+        )
+        if accession is not None:
+            return accession.public_identifier
     if sample.sequencing_reads_collection is not None:
-        for accession in sample.sequencing_reads_collection.accessions:
-            if accession.repository_type == PublicRepositoryType.GISAID:
-                return accession.public_identifier
+        accession = entity_id_to_accession_map.get(
+            sample.sequencing_reads_collection.entity_id, None
+        )
+        if accession is not None:
+            return accession.public_identifier
```

Comment:
this code is really similar to the above block, could probably create another function that returns the accession public_identifier?

Reply:
https://github.com/chanzuckerberg/aspen/pull/232